### ISSUE TITLE
Exposing Options

### ISSUE THREAD

**Issue by srcspider** [sindresorhus/gzip-size#4]
Would be nice if zlib options were exposed.
https://nodejs.org/api/zlib.html#zlib_options

In particular compression level would be nice. I assume since it's not set it's using the default gzip level of `-6`, when what I would like to see is `-9`.

**Comment by sindresorhus** [comment]
I made this to be able to quickly approximate how much gain I would get from gzipping. Not super interested in adding options to it. Would it make sense to default to `-9`? What level do browsers use?

**Comment by srcspider** [comment]
It's a server option; I believe browsers support all levels. I don't ever recall anything about browsers asking for a specific level. In general unless I'm mistaken the recommended setting for production is to have `-9` or level 9 (ie. best possible compression). Default however is for servers like nginx is level 6 (ie. middle of the road); but default for worker processes is also something like 2-4 instead of auto, so it's probably just sensible options for dev boxes.

Reason for having maximum compression, even though the difference is very small a lot of the time, is that processing power is much cheaper then bandwidth. 

I think defaulting to 9 is better then defaulting to 6 since it would then show "just how good it can get" instead of what you would get on non-production settings.

Incidentally, setting that seems to be a bit more troublesome then it would first appear. I've actually tried just adding the options to your code snippet (which according to the node docs should work) but it borks and I'm not sure why.

To get it to work, for my purposes (printing a table of files), I've had to write something like this,

```js
var gulp = require('gulp');
var prettyBytes = require('pretty-bytes');
var zlib = require('zlib');
var glob = require('glob');
var fs = require('fs');
var c = require('chalk');
var _ = require('shadow.lodash');
var Table = require('cli-table');
// ----------------------------------------------------------------------------

var pathRoot = './public/web/';

gulp.task('gzip:info', function (resolve) {

	var table = new Table({
		head: [
			c.reset.bold.underline('file'),
			c.reset.bold.underline('raw'),
			c.reset.bold.underline('gzip')
		]
	});

	console.log('');

	glob(pathRoot + '*.{js,css}', function (err, files) {

		var pending = files.length;
		var checkDone = function () {
			if (pending == 0) {
				console.log(table.toString());
				console.log('');
				resolve();
			}
		}

		_.each(files.sort(), function (filename) {

			var options = {
				level: 9
			};

			// create a new gzip object
			var gzip = zlib.createGzip(options), buffers=[], nread=0;

			gzip.on('error', function(err) {
				gzip.removeAllListeners();
				gzip=null;
				pending -= 1;
				checkDone();
			});

			gzip.on('data', function(chunk) {
				buffers.push(chunk);
				nread += chunk.length;
			});

			gzip.on('end', function() {
				var buffer;
				switch (buffers.length) {
					case 0: // no data.  return empty buffer
						buffer = new Buffer(0);
						break;
					case 1: // only one chunk of data.  return it.
						buffer = buffers[0];
						break;
					default: // concatenate the chunks of data into a single buffer.
						buffer = new Buffer(nread);
						var n = 0;
						buffers.forEach(function(b) {
							var l = b.length;
							b.copy(buffer, n, 0, l);
							n += l;
						});
						break;
				}

				gzip.removeAllListeners();
				gzip=null;

				var data = buffer.toString();

				table.push([ 
					filename.replace(pathRoot, ''),
					c.red(prettyBytes(file.length)),
					c.yellow(prettyBytes(data.length)) 
				]);

				pending -= 1;
				checkDone();
			});

			var file = fs.readFileSync(filename);
			gzip.write(file);
			gzip.end();
		})

		checkDone();
	});

});
```
Which (gulp boilerplating aside) is a bit more verbose then I would like.

ps. not sure if you're using the sync version from browserify for special reasons but there is a sync version in the normal zlib

**Comment by sindresorhus** [comment]
Fixed in 2.0.0 => https://github.com/sindresorhus/gzip-size/commit/4483839ee48957869166dc0c53dc629f89f94093

> ps. not sure if you're using the sync version from browserify for special reasons but there is a sync version in the normal zlib

The sync method was introduced in node 0.12. This module still supports 0.10.

**Comment by sindresorhus** [comment]
Actually. I just dropped node 0.10 compat. Not worth the trouble.
